apps/guardian-ui/src/components/setup/screens/setConfiguration/SetConfiguration.tsx

```diff
@@ -0,0 +1,314 @@
+  const [blockConfirmations, setBlockConfirmations] = useState('');
+  const [network, setNetwork] = useState('');
+  const [bitcoinRpc, setBitcoinRpc] = useState<BitcoinRpc>({
+    kind: '',
+    url: '',
+  });
+  const [mintAmounts, setMintAmounts] = useState<number[]>([]);
+  const [error, setError] = useState<string>();
+  const { onCopy, hasCopied } = useClipboard(password);
+  const [numPeers, setNumPeers] = useState(
+    stateNumPeers ? stateNumPeers.toString() : isSolo ? '1' : MIN_BFT_NUM_PEERS
+  );
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  useEffect(() => {
+    const initStateFromParams = (params: ConfigGenParams) => {
+      setDefaultParams(params);
+      setFederationName(params.meta?.federation_name || '');
+
+      const meta = { federation_name: '', ...params.meta };
+      setMetaFields(Object.entries(meta));
+
+      const mintModule = getModuleParamsFromConfig(params, ModuleKind.Mint);
+      if (mintModule?.consensus?.mint_amounts) {
+        setMintAmounts(mintModule?.consensus.mint_amounts);
+      }
+
+      const walletModule = getModuleParamsFromConfig(params, ModuleKind.Wallet);
+      const walletConsensus = walletModule?.consensus;
+      if (walletConsensus?.finality_delay !== undefined) {
+        setBlockConfirmations(walletConsensus.finality_delay.toString());
+      }
+      if (walletConsensus?.network) {
+        setNetwork(walletConsensus.network);
+      }
+      if (walletModule?.local?.bitcoin_rpc) {
+        setBitcoinRpc(walletModule.local.bitcoin_rpc);
+      }
+    };
+
+    if (configGenParams === null) {
+      api
+        .getDefaultConfigGenParams()
+        .then(initStateFromParams)
+        .catch((err) => {
+          console.error(err);
+        });
+    } else {
+      initStateFromParams(configGenParams);
+    }
+  }, [configGenParams, api]);
+
+  // Update password when updated from state
+  useEffect(() => {
+    setPassword(statePassword);
+  }, [statePassword]);
+
+  const isValid: boolean = isHost
+    ? Boolean(
+        myName &&
+          password &&
+          passwordCheck &&
+          federationName &&
+          isValidNumber(numPeers, 4) &&
+          isValidNumber(blockConfirmations, 1, 200) &&
+          isValidMeta(metaFields) &&
+          network
+      )
+    : isSolo
+    ? Boolean(
+        myName &&
+          password &&
+          passwordCheck &&
+          federationName &&
+          isValidNumber(blockConfirmations, 1, 200) &&
+          isValidMeta(metaFields) &&
+          network
+      )
+    : Boolean(myName && password && hostServerUrl);
+
+  const handleChangeFederationName = (
+    ev: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const newName = ev.currentTarget.value;
+    setFederationName(newName);
+    setMetaFields((prev) =>
+      prev.map(([key, val]) =>
+        key === 'federation_name' ? [key, newName] : [key, val]
+      )
+    );
+  };
+
+  const handleNext = () => {
```

Comment:
nit (perf): Since these functions are being passed down as props, they're getting redefined on every render. 

Whenever a function gets passed in props, the function definition should be wrapped in `useCallback`

Reply:
opening separate issue and will address in separate PR to keep this as refactor.